### PR TITLE
Add aten::trunc, aten::xlogy and thieir variants   

### PR DESCRIPTION
**Pull request by yucai-intel**

### CODE REVIEW

File: yaml/xpu_functions.yaml [hunk 1]
```diff
@@ -45,6 +45,13 @@ supported:
   - remainder_.Tensor
   - remainder.Tensor_out
   - remainder.Scalar_Tensor
+  - xlogy.OutScalar_Self
```
Comment (majing921201) [inline yaml/xpu_functions.yaml:48]: no need to register

File: yaml/xpu_functions.yaml [hunk 2]
```diff
@@ -45,6 +45,13 @@ supported:
...
+  - xlogy.OutScalar_Other
```
Comment (majing921201) [inline yaml/xpu_functions.yaml:49]: ditto

File: yaml/xpu_functions.yaml [hunk 3]
```diff
@@ -45,6 +45,13 @@ supported:
...
+  - xlogy.OutTensor
+  - xlogy.Scalar_Other
```
Comment (majing921201) [inline yaml/xpu_functions.yaml:51]: ditto

File: yaml/xpu_functions.yaml [hunk 4]
```diff
@@ -45,6 +45,13 @@ supported:
...
+  - xlogy.Scalar_Self
```
Comment (majing921201) [inline yaml/xpu_functions.yaml:52]: ditto

File: yaml/xpu_functions.yaml [hunk 5]
```diff
@@ -45,6 +45,13 @@ supported:
...
+  - xlogy.Tensor
+  - xlogy_.Scalar_Other
```
Comment (majing921201) [inline yaml/xpu_functions.yaml:54]: ditto
please double check the auto gen file build/aten/src/ATen/RegisterCUDA.cpp for which operator should be registered
```
m.impl("xlogy.Tensor", TORCH_FN(wrapper_CUDA_xlogy_Tensor));
m.impl("xlogy.OutTensor", TORCH_FN(wrapper_CUDA_xlogy_out_OutTensor));
m.impl("xlogy_.Tensor", TORCH_FN(wrapper_CUDA_xlogy__Tensor));
```

File: src/ATen/native/xpu/BinaryOps.cpp [hunk 6]
```diff
@@ -294,6 +295,50 @@ Tensor XPUNativeFunctions::remainder(const Scalar& self, const Tensor& other) {
...
 
+Tensor XPUNativeFunctions::xlogy(const Tensor& self, const Tensor& other) {
+  Tensor out;
+  TensorIterator iter;
+  iter.build_borrowing_binary_float_op(out, self, other);
+  native::xpu::xlogy_kernel(iter);
+  return iter.output();
+}
+
+Tensor& XPUNativeFunctions::xlogy_out(
```
Comment (majing921201) [inline src/ATen/native/xpu/BinaryOps.cpp:306]: remove corresponding implementation

File: src/ATen/native/xpu/BinaryOps.cpp [hunk 7]
```diff
@@ -294,6 +295,50 @@ Tensor XPUNativeFunctions::remainder(const Scalar& self, const Tensor& other) {
...
+    const Tensor& self,
+    const Tensor& other,
+    Tensor& out) {
+  TensorIterator iter;
+  iter.build_borrowing_binary_float_op(out, self, other);
+  native::xpu::xlogy_kernel(iter);
+  return out;
+}
+
+Tensor& XPUNativeFunctions::xlogy_out(
```
Comment (majing921201) [inline src/ATen/native/xpu/BinaryOps.cpp:316]: ditto

File: src/ATen/native/xpu/BinaryOps.cpp [hunk 8]
```diff
@@ -294,6 +295,50 @@ Tensor XPUNativeFunctions::remainder(const Scalar& self, const Tensor& other) {
...
+    const Scalar& self,
+    const Tensor& other,
+    Tensor& result) {
+  return at::xlogy_out(result, native::wrapped_scalar_tensor(self), other);
+}
+
+Tensor& XPUNativeFunctions::xlogy_out(
```
Comment (majing921201) [inline src/ATen/native/xpu/BinaryOps.cpp:323]: ditto

File: src/ATen/native/xpu/BinaryOps.cpp [hunk 9]
```diff
@@ -294,6 +295,50 @@ Tensor XPUNativeFunctions::remainder(const Scalar& self, const Tensor& other) {
...
+    const Tensor& self,
+    const Scalar& other,
+    Tensor& result) {
+  return at::xlogy_out(result, self, native::wrapped_scalar_tensor(other));
+}
+
+Tensor XPUNativeFunctions::xlogy(const Scalar& x, const Tensor& y) {
```
Comment (majing921201) [inline src/ATen/native/xpu/BinaryOps.cpp:330]: ditto

File: src/ATen/native/xpu/BinaryOps.cpp [hunk 10]
```diff
@@ -294,6 +295,50 @@ Tensor XPUNativeFunctions::remainder(const Scalar& self, const Tensor& other) {
...
+  return at::xlogy(native::wrapped_scalar_tensor(x), y);
+}
+
+Tensor XPUNativeFunctions::xlogy(const Tensor& x, const Scalar& y) {
```
Comment (majing921201) [inline src/ATen/native/xpu/BinaryOps.cpp:334]: ditto

File: src/ATen/native/xpu/ReduceOps.cpp [hunk 11]
```diff
@@ -485,7 +485,7 @@ Tensor XPUNativeFunctions::any(const Tensor& self) {
   return out;
 }
 
-// aten::any.any_out
+// aten::any.all_out
```
Comment (majing921201) [inline src/ATen/native/xpu/ReduceOps.cpp:488]: Do we need this change ?

Comment (yucai-intel) [reply src/ATen/native/xpu/ReduceOps.cpp:488]: We have no op named any.any, its actually any.all.

File: yaml/xpu_functions.yaml [hunk 12]
```diff
@@ -36,6 +36,9 @@ supported:
   - remainder_.Tensor
   - remainder.Tensor_out
   - remainder.Scalar_Tensor
+  - xlogy.Tensor
```
Comment (fengyuan14) [inline yaml/xpu_functions.yaml:39]: Please remove these lines and add schemas in `native_functions.yaml` only.

File: src/ATen/native/xpu/BinaryOps.cpp [hunk 13]
```diff
@@ -51,6 +52,7 @@ REGISTER_XPU_DISPATCH(fmax_stub, &xpu::fmax_kernel);
 REGISTER_XPU_DISPATCH(fmin_stub, &xpu::fmin_kernel);
 REGISTER_XPU_DISPATCH(lshift_stub, &xpu::lshift_kernel);
 REGISTER_XPU_DISPATCH(rshift_stub, &xpu::rshift_kernel);
+REGISTER_XPU_DISPATCH(xlogy, &xpu::xlogy_kernel);
```
Comment (fengyuan14) [inline src/ATen/native/xpu/BinaryOps.cpp:55]: xlogy_stub?

File: yaml/xpu_functions.yaml (hunk 12 above)
Comment (yucai-intel) [reply yaml/xpu_functions.yaml]: Done.

File: src/ATen/native/xpu/BinaryOps.cpp (hunk 13 above)
Comment (yucai-intel) [reply src/ATen/native/xpu/BinaryOps.cpp]: Done.

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**